src/kx122/kx122.c

```diff
@@ -205,13 +211,20 @@ kx122_context kx122_init(int bus, int addr, int chip_select_pin)
     mraa_gpio_dir(dev->chip_select,MRAA_GPIO_OUT);
     mraa_spi_mode(dev->spi,MRAA_SPI_MODE0);
 
-    if (mraa_spi_frequency(dev->spi,SPI_FREQUENCY)){
+    if (mraa_spi_frequency(dev->spi, bus_frequency)){
       printf("%s: mraa_spi_frequency() failed.\n", __FUNCTION__);
       kx122_close(dev);
       return NULL;
     }
   }
   else{ //Using I2C
+
+    if (bus_frequency > 3400000){	// KX122 has a maximum I2C bus speed of 3.4MHz
+      printf("%s: bus frequency too high.\n", __FUNCTION__);
```

Comment:
I like the comment text "KX122 has a maximum I2C bus speed of 3.4MHz" vs. the printf text since it's more descriptive.